### ISSUE TITLE
URL field is not showing error messages for x-frame options

### ISSUE THREAD

**Issue by sheadarlison** [Rise-Vision/widget-web-page#124]
https://www.screencast.com/t/VnIe8i54MS9

example problem url: https://codepen.io/dilums/live/mdrEOBy

**Comment by stulees** [comment]
@sheadarlison @olegrise @alanclayton I've done an investigation into this and here's the summary:

- In the settings of this widget we use our [proxy server](https://github.com/Rise-Vision/proxy-server) to request a web page in order to check the response for `X-Frame-Options`
- The proxy server uses [cors-anywhere](https://www.npmjs.com/package/cors-anywhere) nodejs library to facilitate the proxy implementation which adds CORS headers to the proxied request. 
- A limitation of cors-anywhere (and safe practice of proxying requests) is it restricts `cookie` headers as requesting user credentials is disallowed. Therefore `cookie` headers are removed from the request headers, see [here](https://github.com/Rise-Vision/proxy-server/blob/master/app.js#L9)
- The codepen url used in this issue requires cookies, see https://www.screencast.com/t/7qX5wmb4
- When our proxy server makes the request, a 403 response is returned from codepen. 
- Since the widget settings is only checking for the presence of `X-Frame-Options`, it does not display anything in this situation as it did not receive a successful response. 

In conclusion, this is a limitation of the Widget. We can not inform users of `X-Frame-Options` in the widget settings if the web page being used requires cookies. Otherwise, the widget is correctly displaying an error message to user when the page has `X-Frame-Options`, see [here](https://www.screencast.com/t/TZcqqHLGplrD). 

I recommend we close this issue. Thanks

**Comment by alanclayton** [comment]
@stulees I agree with your recommendation. @sheadarlison let us know if you agree. Thanks

**Comment by stulees** [comment]
@sheadarlison Going ahead and closing this, let us know if you need further discussion, thanks.
